### ISSUE TITLE
Feature request: Smarter cropping with face detection

### ISSUE THREAD

**Issue by sh95014** [Dimillian/IceCubesApp#1762]
Web:
<img width="569" alt="web" src="https://github.com/Dimillian/IceCubesApp/assets/95387068/f618c7b1-637a-4229-97c9-e0d506dc5aa4">

IceCubes:
![icecubes](https://github.com/Dimillian/IceCubesApp/assets/95387068/176b00be-7634-4796-8e7e-011dfee9efef)

A relatively cheap way of improving this might be:

1. extract downloaded image from the `LazyResizableImage`
2. detect faces in the image using Vision.framework
3. take the average vertical position of the detected faces
4. set the `frame(alignment:)` of the image in `defaultLinkPreview` to whichever is closest to the average vertical position of faces (so `.top` in the example above)

Unfortunately I can't figure out step 1 and dropping to the lower-level code of Nuke seems like a disproportionate amount of work for the benefit. There's `ImageRenderer` but that's gross.

Filing here anyway in case anybody has other ideas.

**Comment by Dimillian** [comment]
When scrolling at 60/120 FPS not sure it's really worth it. Would sure be nice but not sure at all it's worth it indeed.

**Comment by sh95014** [comment]
Yeah, probably better with server support, rather than doing client-side detection.
